gst/tensor_filter/tensor_filter.c

```diff
@@ -353,6 +355,27 @@ gst_tensor_filter_init (GstTensorFilter * self)
   gst_tensors_config_init (&self->out_config);
 }
 
+/**
+ * @brief Function to finalize instance.
+ */
+static void
+gst_tensor_filter_finalize (GObject * object)
+{
+  GstTensorFilter *self;
+  GstTensorFilterProperties *prop;
+
+  self = GST_TENSOR_FILTER (object);
+  prop = &self->prop;
+
+  self->configured = FALSE;
+  gst_tensors_config_init (&self->in_config);
+  gst_tensors_config_init (&self->out_config);
+
+  g_free ((char *) prop->model_file);
+
+  G_OBJECT_CLASS (parent_class)->finalize (object);
+}
```

Comment:
Finalize is called when the plugin (gobject) is destroyed, not when an instance of the plugin (an element in a pipeline) is destroyed. Is this really what you want?